src/forms/FormikSwitchField.tsx

```diff
@@ -0,0 +1,126 @@
+import * as RadixSwitch from '@radix-ui/react-switch'
+import { useField } from 'formik'
+import React, { useContext } from 'react'
+import { twMerge } from 'tailwind-merge'
+import Label from '../forms/Label'
+import { ThemeContext } from '../ThemeProvider'
+
+export interface FormikSwitchFieldProps {
+  id?: string
+  name: string
+  data?: {
+    cy?: string
+    test?: string
+  }
+  disabled?: boolean
+  label?: string
+  fluid?: boolean
+  labelLeft?: boolean
+  size?: 'sm' | 'md' | 'lg'
+  className?: {
+    root?: string
+    element?: string
+    thumb?: string
+    label?: string
+  }
+}
+
+const defaultProps = {
+  id: undefined,
+  data: undefined,
+  className: undefined,
+  disabled: false,
+  label: undefined,
+  labelLeft: false,
+  fluid: false,
+  size: 'md',
+}
+
+/**
+ * This function returns a pre-styled Switch component based on the RadixUI switch component and the custom theme that works as to be expected in a Formi environment.
+ * State is managed by Formik through the name attribute.
+ *
+ * @param id - The id of the switch.
+ * @param name - The name of the field. This is used to identify the field in Formik.
+ * @param data - The object of data attributes that can be used for testing (e.g. data-test or data-cy)
+ * @param label - The label that is displayed next to the switch.
+ * @param disabled - Indicator whether the switch is disabled or not.
+ * @param fluid - Indicator whether the switch should be fluid or not.
+ * @param labelLeft - Indicator whether the label should be displayed on the left or right side of the switch.
+ * @param size - The size of the switch. The size can be small, medium or large.
+ * @param className - The optional className object allows you to override the default styling.
+ * @returns Switch component with formik state management
+ */
+export function FormikSwitchField({
+  id,
+  name,
+  data,
+  disabled,
+  label,
+  fluid,
+  labelLeft,
+  size,
+  className,
+}: FormikSwitchFieldProps) {
+  const theme = useContext(ThemeContext)
+  const [field, meta, helpers] = useField(name)
+
+  const rootSize = {
+    sm: 'w-10 h-[1.3rem]',
+    md: 'w-12 h-[1.6rem]',
+    lg: 'w-16 h-[1.9rem]',
+  }
+  const thumbSize = {
+    sm: 'ml-[0.2rem] w-4 h-4',
+    md: 'ml-1 w-5 h-5',
+    lg: 'ml-1 w-6 h-6',
+  }
+  const transitionSize = {
+    sm: 'translate-x-[1.1rem]',
+    md: 'translate-x-[1.25rem]',
+    lg: 'translate-x-[2rem]',
+  }
+
+  return (
+    <div
+      className={twMerge(
+        'flex flex-row items-center gap-3',
+        fluid && 'justify-between',
+        className?.root
+      )}
+    >
+      {labelLeft && label && (
+        <Label className={{ root: className?.label }} label={label} />
+      )}
+      <RadixSwitch.Root
```

Comment:
@BShaq We probably shouldn't copy the entire switch component from Switch.tsx to this file, just to create a Formik component with corresponding state handling from it (and probably this shouldn't be necessary either, right?). I think we should try to just import the Switch component and use it with the formik values and states. Otherwise, we would have to copy every change that we make to the Switch component to this file as well...

Reply:
Thanks for the update! Would is maybe also make sense to drop certain attributes that will not be used in connection with the Switch used as a field, like the `fluid` prop or maybe even `labelLeft`?

Reply:
Yeah, good point 👍  I reused the component now. Together with @rschlaefli , we figured out yesterday that letting formik handle the state makes it a lot easier in the case of the creation wizard, that's why we wanted to introduce this component

Reply:
Sure, I'll address it :)